client/data/deposits/test/resolvers.js

```diff
@@ -101,6 +121,9 @@ describe( 'getDeposits resolver', () => {
 		'page=1&pagesize=25&match=all&store_currency_is=gbp&date_before=2020-04-29%2003%3A59%3A59&date_after=2020-04-29%2004%3A00%3A00&date_between%5B0%5D=2020-04-28%2004%3A00%3A00&date_between%5B1%5D=2020-04-30%2003%3A59%3A59&status_is=paid&status_is_not=failed';
 
 	beforeEach( () => {
+		apiFetch.mockImplementation( () => {
```

Comment:
To be honest, I don't know why this is necessary but without it, I get this error:

```
expect(received).toEqual(expected) // deep equality

Expected: undefined
Received: {}
```

as if `generator.next().value` is `{}` and the return value of mocked `apiFetch()` is `undefined`.

I'm not surprised that the return value of mocked `apiFetch()` is `undefined` because I did not specify the mocked function to return anything but I'm surprised why `generator.next().value` is `{}`.

Reply:
Might be worth @Jinksi taking a look at this?

Reply:
I probably need to take a deeper look, but it could be due to this line, as apiFetch is mocked and `undefined`, it resorts to `{}` 

https://github.com/Automattic/woocommerce-payments/blob/9a894edda84e2e7d77b3e9fdaf63e6153acce1c2/client/data/deposits/resolvers.js#L114

Reply:
Btw if you want to avoid this change, you can also return a mock value at the top.

```js
jest.mock( '@wordpress/data-controls', () => ( {
	apiFetch: jest.fn().mockReturnValue( 'something' ),
} ) );
```

Reply:
Thank you for the explanation, @naman03malhotra. That totally makes sense!

However, since only `getDeposits()` that defaults to `{}` when `apiFetch()` returns `undefined` (and that explains why only the test suite for `getDeposits()` that fails after I mock `apiFetch()`), I think it makes sense to just mock the return value in the `beforeEach()` of the `getDeposits()`'s test suite, which is already done in this PR.

Since there is no more mystery, I'm merging this PR.